### PR TITLE
Add BMS service interfaces

### PR DESCRIPTION
**Pull request by disha-wani**
**What this PR does / why we need it**: This PR adds BMS interfaces

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**: NONE

**Release note**: NONE

```release-note
```

### CODE REVIEW

File: openstack/bms/v2/servers/requests.go [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+func getStructServerField(v *Server, field string) string {
+	r := reflect.ValueOf(v)
+	f := reflect.Indirect(r).FieldByName(field)
+	return string(f.String())
+}
+
+// Get requests details on a single server, by ID.
+func Get(client *golangsdk.ServiceClient, id string) (r GetResult) {
+	_, r.Err = client.Get(getURL(client, id), &r.Body, &golangsdk.RequestOpts{
+		OkCodes:     []int{200},
```
Comment (freesky-edward) [inline openstack/bms/v2/servers/requests.go]: the default ok code of ```Get``` is 200,  it isn't necessary to list here. please refer to https://github.com/huaweicloud/golangsdk/blob/aef01041fa9cf1fa8844a490cac4ded3fceec9a7/provider_client.go#L356:2

File: openstack/bms/v2/common/common_tests.go [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+	"github.com/huaweicloud/golangsdk"
+	"github.com/huaweicloud/golangsdk/testhelper/client"
+)
+
+const TokenID = client.TokenID
+
+func ServiceClient() *golangsdk.ServiceClient {
+	sc := client.ServiceClient()
+	e := strings.Replace(sc.Endpoint, "v2", "v2.1", 1)
```
[the remark this answers is not in the document]
Comment (Savasw) [reply openstack/bms/v2/common/common_tests.go]: @freesky-edward can you please elaborate your question

[the remark this answers is not in the document]
Comment (freesky-edward) [reply openstack/bms/v2/common/common_tests.go]: I mean if any case will use the special ```endpoint``` as a case to test, if not, why replace the version?

[the remark this answers is not in the document]
Comment (disha-wani) [reply openstack/bms/v2/common/common_tests.go]: @freesky-edward we have a fake client in testhelper, which came to my notice, which is directly used here so do not need this file, hence deleting this.

**Comment by freesky-edward** [comment]
LGTM

File: openstack/bms/v2/common/common_tests.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (freesky-edward) [reply openstack/bms/v2/common/common_tests.go]: thanks, we are going to remove all the replacement code like this party. that is why I ask this question.

**Pull request merged**